doc/source/whatsnew/v3.0.0.rst

```diff
@@ -419,6 +419,7 @@ Interval
 Indexing
 ^^^^^^^^
 - Bug in :meth:`DataFrame.__getitem__` returning modified columns when called with ``slice`` in Python 3.12 (:issue:`57500`)
+- Bug in :meth:`DatetimeIndex.is_year_start` and :meth:`DatetimeIndex.is_quarter_start` returning ``False`` on double-digit frequencies (:issue:`58523`)
```

Comment:
this should have been in `Datetimelike`, "indexing" is more like `.loc` / `.iloc` / get/setitem stuff

but OK to address this as part of https://github.com/pandas-dev/pandas/pull/58665/files, as that one needs updating anyway

Reply:
thanks, I addressed this comment in the PR you suggested